src/http.rs

```diff
@@ -45,77 +46,151 @@ async fn handle_web_req(
+}
 
-        (&Method::POST, "/pj") => {
-            dbg!(req.uri().query());
-
-            let headers = Headers(req.headers().to_owned());
-            let query = {
-                let uri = req.uri();
-                if let Some(query) = uri.query() {
-                    Some(&query.to_owned());
-                }
-                None
-            };
-            let body = req.into_body();
-            let bytes = hyper::body::to_bytes(body).await?;
-            dbg!(&bytes); // this is correct by my accounts
-            let reader = &*bytes;
-            let original_request = UncheckedProposal::from_request(reader, query, headers).unwrap();
-
-            let proposal_psbt = scheduler.propose_payjoin(original_request).await.unwrap();
-
-            Ok(Response::new(Body::from(proposal_psbt)))
-        }
+async fn handle_static(path: &str) -> Result<Response<Body>, HttpError> {
+    let directory_traversal_vulnerable_path = &path[("/pj/static/".len())..];
+    match std::fs::read(Path::new(STATIC_DIR).join(directory_traversal_vulnerable_path)) {
+        Ok(file) => Response::builder()
+            .status(200)
+            .header("Cache-Control", "max-age=604800")
+            .body(Body::from(file))
+            .map_err(HttpError::Http),
+        Err(_) => handle_404().await,
+    }
+}
 
-        (&Method::POST, "/pj/schedule") => {
-            let bytes = hyper::body::to_bytes(req.into_body()).await?;
-            // deserialize x-www-form-urlencoded data with non-strict encoded "channel[arrayindex]"
-            let conf = serde_qs::Config::new(2, false);
-            let request: ScheduledPayJoin =
-                conf.deserialize_bytes(&bytes).expect("invalid request");
-
-            let address = scheduler.schedule_payjoin(&request).await.unwrap();
-            let total_amount = request.total_amount();
-            let uri = scheduler::format_bip21(address.clone(), total_amount, endpoint);
-            let mut response = Response::new(Body::from(uri.clone()));
-            create_qr_code(&uri, &address.to_string());
-            response
-                .headers_mut()
-                .insert(hyper::header::CONTENT_TYPE, "text/plain".parse().unwrap());
-            Ok(response)
-        }
+async fn handle_pj(scheduler: Scheduler, req: Request<Body>) -> Result<Response<Body>, HttpError> {
+    dbg!(req.uri().query());
 
-        // Return the 404 Not Found for other routes.
-        _ => {
-            let mut not_found = Response::default();
-            *not_found.status_mut() = StatusCode::NOT_FOUND;
-            Ok(not_found)
+    let headers = Headers(req.headers().to_owned());
+    let query = {
+        let uri = req.uri();
+        if let Some(query) = uri.query() {
+            Some(&query.to_owned());
         }
-    }
+        None
+    };
+    let body = req.into_body();
+    let bytes = hyper::body::to_bytes(body).await?;
+    dbg!(&bytes); // this is correct by my accounts
+    let reader = &*bytes;
```

Comment:
Do we actually want to dbg! these bytes?

Reply:
It has let us check the original PSBT in the past. Of little use here. We could introduce the log crate to show debug/warn/errors logging soon

Reply:
removed it for this pr